connman/plugins/wifi.c

```diff
@@ -769,6 +779,22 @@ static void scan_callback_hidden(int result,
 	scan_callback(result, interface, user_data);
 }
 
+static gboolean carrier_timeout(gpointer data)
+{
+	struct wifi_data *wifi = data;
+
+	if (!wifi)
```

Comment:
Need to zero `wifi->carrier_timeout` so that we don't try to remove non-existent source later.